lookout/style/typos/analyzer.py

```diff
@@ -128,34 +136,85 @@ def generate_typos_fixes(self, changes: Sequence[Change]) -> Iterator[TypoFix]:
-                        )
-
-    def render_comment_text(self, typo_fix: TypoFix) -> str:
+                    yield TypoFix(
+                        head_file=file,
+                        identifier=new_identifiers[index].token,
+                        candidates={token: [Candidate(*sugg) for sugg in suggestions[index][token]]
+                                    for token in suggestions[index]},
+                        line_number=new_identifiers[index].start_position.line,
+                    )
+
+    def render_comment_text(self, typo_fix: TypoFix) -> (str, Iterable[float]):
         """
         Generate the text of the comment for the specified typo fix.
 
         :param typo_fix: Information about typo fix required to render a comment text.
         :return: string with the generated comment.
         """
-        # TODO: move this logic to template. See FormatAnalyzer.render_comment_text()
-        corrections_line = ", ".join("%s (%d%%)" % (
-            c.token, int(c.confidence * 100)) for c in typo_fix.candidates)
-        return "Possible typo in \"%s\". Suggestions: %s" % (typo_fix.token, corrections_line)
+        file_lines = typo_fix.head_file.content.decode("utf-8", "replace").splitlines()
+        suggestions, confidences = [], []
+        for final_sugg, conf in self.generate_identifier_suggestions(
+                typo_fix.candidates, typo_fix.identifier,
+        ):
+            suggestions.append(final_sugg)
+            confidences.append(conf)
+        return self.comment_template.render(
+            identifier=typo_fix.identifier,
+            suggestions=suggestions,
+            old_code_line=file_lines[typo_fix.line_number - 1],
+            confidences=self._normalize_confidences(confidences),
+            zip=zip,
+            enumerate=enumerate,
+            int=int,
+        ), confidences
+
+    @staticmethod
+    def _normalize_confidences(confidences: Sequence[float]):
+        s = sum(confidences)
+        return [conf / s for conf in confidences]
+
+    def generate_identifier_suggestions(self, suggestions: Mapping[str, Iterable[Candidate]],
+                                        identifier: str) \
+            -> Iterator[Tuple[str, float]]:
+        """
+        Generate suggestions for the identifier and compute the probability of suggestion.
+
+        :param suggestions: suggestions are a mapping from a token to the list of candidates.
+        :param identifier: initial identifier.
+        :return: a generator of tuples with a suggestion for the identifier and probability.
+        """
+        if not suggestions:
+            # do nothing in case of empty suggestions
+            return
+        token_candidates = []
+        # split identifier
+        initial_tokens = self.parser.split(identifier)
+
+        # prepare suggestions per initial token
+        for t in initial_tokens:
+            try:
+                token_candidates.append(suggestions[t])
+            except KeyError:
```

Comment:
Btw, you can avoid it setting `return_all=True` in TyposCorrector calling [here](https://github.com/src-d/style-analyzer/pull/709/files#diff-18972d7c14aaaa30c48c1560b2e149c0R295). This way suggestions will contain all tokens, and the ones that are considered not-typoed will have candidates in this form: `A: [A, 1.0]` - exactly what you do here. In your way there are much less information passed, so maybe it's better to leave it as it is, idk.